src/app/shared/form/builder/parsers/field-parser.ts

```diff
@@ -344,7 +346,13 @@ export abstract class FieldParser {
   }
 
   protected addPatternValidator(controlModel) {
-    const regex = new RegExp(this.configData.input.regex);
+    const validatorMatcher = this.configData.input.regex.match(REGEX_FIELD_VALIDATOR);
```

Comment:
Can we add some comments here to describe what this is validating?   Currently, this code is hard to read/understand because it is not clear what the `REGEX_FIELD_VALIDATOR` is doing.   It's very important to add comments to complex regex so that developers understand their purpose.

Alternatively, you could add some specs which prove that this validator is working as expected.